### PR TITLE
making `columns` works for Page and Resource

### PR DESCRIPTION
**Pull request by bsvin33t**
Calculating the column styles and margins only if the child element is of type column.

### CODE REVIEW

File: lib/active_admin/views/components/columns.rb [hunk 1]
```diff
@@ -93,15 +93,21 @@ def calculate_columns!
...
       end
 
       def columns_span_count
         count = 0
-        children.each {|column| count += column.span_size }
+        children.each do|child|
+          if child.is_a?(Column)
+            count += child.span_size
+          end
+        end
```
Comment (timoschilling) [inline lib/active_admin/views/components/columns.rb]: You can simplify Line 97 and Line 107 by building a `columns` method, wich looks like:
```ruby
def columns
  children.select { |child| child.is_a?(Column) } 
end
```

**Comment by timoschilling** [comment]
Good work. Please apply the Hound comments.

**Comment by timoschilling** [comment]
There is no need to merge / pick the master into your PR. Please rebase your PR on master and squash all commits.

File: lib/active_admin/views/components/columns.rb [hunk 2]
```diff
@@ -93,19 +93,24 @@ def calculate_columns!
         all_margins_width = margin_size * (span_count - 1)
         column_width = (100.00 - all_margins_width) / span_count
 
-        children.each_with_index do |col, i|
+        columns.each_with_index do |column, i|
           is_last_column = i == (columns_count - 1)
-          col.set_column_styles(column_width, margin_size, is_last_column)
+          column.set_column_styles(column_width, margin_size, is_last_column)
```
Comment (timoschilling) [inline lib/active_admin/views/components/columns.rb]: I'm not sure that this code does the same as the old one. What `children` can be there that are not a `Column`?

Comment (bsvin33t) [inline lib/active_admin/views/components/columns.rb]: maybe something like a `para` ?

Comment (timoschilling) [inline lib/active_admin/views/components/columns.rb]: I don't know, but if the tests are working keep it as it is.

**Pull request merged**